lib/internal/main/eval_string.js

```diff
@@ -23,6 +23,17 @@ const {
 const { addBuiltinLibsToObject } = require('internal/modules/helpers');
 const { getOptionValue } = require('internal/options');
 
+const {
+  codes: {
+    ERR_CODE_INTEGRITY_BLOCKED,
+  },
+} = require('internal/errors');
+
+const ci = require('internal/code_integrity');
+if (ci.isInteractiveModeDisabled()) {
+  throw new ERR_CODE_INTEGRITY_BLOCKED('"eval"');
+}
```

Comment:
This should be surrounded by isWindows check and we shouldn't even require this file for non-windows environments